modPrecompiled/src/org/aion/precompiled/PrecompiledResultCode.java

```diff
@@ -0,0 +1,96 @@
+package org.aion.precompiled;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.aion.vm.api.interfaces.ResultCode;
+
+/**
+ * An enumeration representing the execution status of a transaction.
+ */
+public enum PrecompiledResultCode implements ResultCode {
```

Comment:
How come FastVm, Avm, and Precompiled all have their own ResultCode class?  I worry that this will lead us down a path where different VMs give different result codes for the same result / a situation where the caller has to know about the underlying VM in order to interpret the result code.  Can we make them use the same set of result codes?

Reply:
This is one of the things that is still in progress.
We are considering making everything except `FAILURE` the same and put into the API and allowing a VM to declare their own `FAILURE` types, since these are run-time failures, which may differ between VM's.
Not sure how we will handle this yet, but the issue is known